age/src/ssh/identity.rs

```diff
@@ -310,15 +310,18 @@ fn rsa_privkey(input: &str) -> IResult<&str, Identity> {
                     if enc_header.is_some() {
                         Some(UnsupportedKey::EncryptedPem.into())
                     } else {
-                        read_asn1::rsa_privkey(&privkey).ok().map(|(_, privkey)| {
-                            let mut ssh_key = vec![];
-                            cookie_factory::gen(
-                                write_ssh::rsa_pubkey(&privkey.to_public_key()),
-                                &mut ssh_key,
-                            )
-                            .expect("can write into a Vec");
-                            UnencryptedKey::SshRsa(ssh_key, Box::new(privkey)).into()
-                        })
+                        use rsa::pkcs1::FromRsaPrivateKey;
+                        rsa::RsaPrivateKey::from_pkcs1_der(&privkey)
```

Comment:
> If you parse the PEM label with pem-rfc7468 and you get `ENCRYPTED RSA PRIVATE KEY` is that sufficient to indicate to the user that this is an unsupported key file?

No, because `ENCRYPTED RSA PRIVATE KEY` is not what is used as the label in practice; you instead get the same PEM label as for unencrypted, but with the additional header. Here's what I get from `openssl 1.1.1f`:

```
$ openssl genrsa -out example.key 2048
Generating RSA private key, 2048 bit long modulus (2 primes)
...............+++++
....................+++++
e is 65537 (0x010001)
$ cat example.key | head -n 2
-----BEGIN RSA PRIVATE KEY-----
MIIEogIBAAKCAQEA9WDz0dl/aYSgxHso3rt9c1ljZ1qvmOUOCa1oyX0dAX3YQqlF
$ openssl rsa -des3 -in example.key -out protected.key
writing RSA key
Enter PEM pass phrase:
Verifying - Enter PEM pass phrase:
$ cat protected.key | head -n 5
-----BEGIN RSA PRIVATE KEY-----
Proc-Type: 4,ENCRYPTED
DEK-Info: DES-EDE3-CBC,9BC872773CF8C93B

U8d2d6DqZNX20UyrOr5fqc+eJCaOzj8G8fPm4gdmvESVzAW95BNQpRHodAEW6Nx5
```

I want to continue to distinguish three cases, as I do in the current parser:
- Data has no PEM header.
  - Great, we carry on parsing, and maybe return a supported key!
- Data has a PEM header matching the structure of the encrypted PEM header.
  - We stop parsing and return `Identity::Unsupported(UnsupportedKey::EncryptedPem)`.
- Data has an unknown PEM header.
  - We return a parser error.

Per my comment in https://github.com/RustCrypto/formats/issues/112#issuecomment-945161674, I think how we could achieve this is to match on the `pem_rfc7468::Error::HeaderDisallowed` error, and in that case we re-parse the data ourselves with the following small parser:
```rust
preceded(
    tuple((
        tag("-----BEGIN "),
        tag(pem_label),
        tag("-----"),
        line_ending,
    )),
    terminated(rsa_pem_encryption_header, line_ending),
)
```

This would result in the PEM header being parsed three separate times (once to get `Error::HeaderDisallowed`, once to get `pem_label`, once to detect the encryption header), but this would only happen in the error case where a header exists, so it shouldn't add any parsing cost to the hot path.

Reply:
I'm going to leave this code as-is for now.